src/test/scala/sigmastate/serialization/DataSerializerSpecification.scala

```diff
@@ -0,0 +1,33 @@
+package sigmastate.serialization
+
+import java.math.BigInteger
+import java.nio.ByteBuffer
+
+import sigmastate.SCollection.SByteArray
+import sigmastate.interpreter.CryptoConstants.EcPointType
+import sigmastate._
+import sigmastate.utils.ByteArrayBuilder
+import sigmastate.utxo.ErgoBox
+
+class DataSerializerSpecification extends SerializationSpecification {
+
+  def roundtrip[T <: SType](obj: T#WrappedType, tpe: T) = {
+    val b = new ByteArrayBuilder()
+    DataSerializer.serialize(obj, tpe, b)
+    val buf = ByteBuffer.wrap(b.toBytes)
+    val res = DataSerializer.deserialize(tpe, buf)
+    res shouldBe obj
+  }
+
+  property("Data serialization round trip") {
+    forAll { x: Byte => roundtrip[SByte.type](x, SByte) }
+    forAll { x: Boolean => roundtrip[SBoolean.type](x, SBoolean) }
+    forAll { x: Long => roundtrip[SInt.type](x, SInt) }
+    forAll { x: BigInteger => roundtrip[SBigInt.type](x, SBigInt) }
+    forAll { x: EcPointType => roundtrip[SGroupElement.type](x, SGroupElement) }
+    forAll { x: ErgoBox => roundtrip[SBox.type](x, SBox) }
```

Comment:
Neat! I guess this makes `ErgoBoxSerializerSpec` obsolete, right?

Reply:
Yes, ones this will be finished. Here is also an example using ByteBuffer and ByteArrayBuilder as more efficient and compact way to do serialization.